### ISSUE TITLE
Bug: input type month and VoiceOver

### ISSUE THREAD

**Issue by wtchnm** [facebook/react#23131]
`<input type="month" />` does not work correctly with iOS VoiceOver enabled.

## Steps To Reproduce

1. Open https://3j4l9.csb.app/
2. Activate VoiceOver
3. Double click in the label

## The current behavior
- CodeSandbox: https://codesandbox.io/s/react-month-input-bug-3j4l9?file=/public/index.html:0-357
- Image:
<img width="414" alt="React App" src="https://user-images.githubusercontent.com/37726261/149831927-fbc783b0-77e9-4601-b143-e09834ca8727.png">

## The expected behavior
- CodeSandbox: https://codesandbox.io/s/react-month-input-bug-vanilla-dcg6d?file=/index.html
- Image:
<img width="414" alt="Vanilla App" src="https://user-images.githubusercontent.com/37726261/149831910-ecea3120-9083-4ba1-91f0-878dbd2ffb61.png">

**Comment by wtchnm** [comment]
Bump
